src/Workspaces/Core/Portable/Shared/TestHooks/AsynchronousOperationListener.cs

```diff
@@ -106,7 +105,12 @@ public Task CreateWaitTask()
                 }
                 else
                 {
-                    var source = new TaskCompletionSource<bool>();
+                    // setting result of a task can cause await machinary to wake up awaited code
```

Comment:
Unsure if this comment is now needed, since you're just explaining the flag.

Reply:
(or the comment is stale now)

Reply:
I believe the comment is still helpful. this is doc on the option - https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.taskcreationoptions?view=netframework-4.7.2#System_Threading_Tasks_TaskCreationOptions_RunContinuationsAsynchronously

which explains what it does, but it is not clear why we need to use it here.

Reply:
i agree with Heejae.  The comment is helpful for making it clearer that it's intentional and important htat RunContinuationsAsynchronously be called.

Reply:
@heejaechang You might consider tweaking the comment to:

Calling SetResult on a normal TaskCompletionSource can cause continuations to run synchronously at that point.  That's a problem as that may cause additional code to run while we're holding a lock.  In order to prevent that, we pass along `RunContinuationsAsynchronously` in order to ensure that all continuations will run at a future point when this thread has released the lock.

Reply:
sure. thank you